pkg/kv/kvserver/batcheval/cmd_scan_test.go

```diff
@@ -231,6 +232,154 @@ func TestScanReverseScanWholeRows(t *testing.T) {
 	})
 }
 
+// TestScanReturnRawMVCC tests that a ScanRequest with the
+// ReturnRawMVCC option set should return the full bytes of the
+// MVCCValue in the RawBytes field of returned roachpb.Values.
+func TestScanReturnRawMVCC(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	ctx := context.Background()
+
+	db := storage.NewDefaultInMemForTesting()
+	defer db.Close()
+
+	// Write a value with WriteOptions that will produce an
+	// MVCCValueHeader.
+	clock := hlc.NewClockForTesting(nil)
+	settings := cluster.MakeTestingClusterSettings()
+	evalCtx := (&MockEvalCtx{Clock: clock, ClusterSettings: settings}).EvalContext()
+	ts := clock.Now()
+	txn := roachpb.MakeTransaction("test", nil /* baseKey */, isolation.Serializable, roachpb.NormalUserPriority, ts, 0, 0, 0, false /* omitInRangefeeds */)
+	putWithWriteOptions := func(key roachpb.Key, value roachpb.Value, writeOpts *kvpb.WriteOptions) {
```

Comment:
am I seeing correctly that all writes in this test are intents? Can you add one that isn't and make sure this also works? I don't expect any surprises given your testing of pebbleMVCCScanner, but still, good coverage to have.

Reply:
Added a get-based test. I also added a case that makes sure we hit both the seek and iter path.